microservices/ingestservice/src/main.py

```diff
@@ -101,27 +137,84 @@ def ingest_item(
         raise HTTPException(status_code=422, detail=error_message)
 
 
-@router.post("/ingest-multiple-items")
-def bulk_ingest(
+@router.post("/ingest-multiple-items", status_code=202)
+def ingest_multiple_items(
     body: FullLoadRequest,
     storage: Annotated[storage.Client, Depends(get_storage_client)],
-):
+    tasks: BackgroundTasks,
+) -> FullLoadResponse:
+    task_id = str(uuid.uuid4())
     bucket = storage.bucket(body.bucket)
-    item_dict = {}
-    for blob in bucket.list_blobs(match_glob=f"{body.prefix}*.json"):
-        logger.info(f"Downloading {blob.name}")
-        data = json.loads(blob.download_as_text())
-        try:
-            mapped_data = data_preprocessor.map_data(data)
+    tasks.add_task(bulk_ingest, bucket, body.prefix, task_id)
+    return FullLoadResponse(task_id=task_id)
+
+
+def bulk_ingest(
+    bucket: storage.Bucket,
+    prefix: str,
+    task_id: str,
+):
+    logger.info("Starting bulk ingest")
+    bulk_ingest_tasks[task_id] = BulkIngestTask(
+        id=task_id, status=BulkIngestTaskStatus.PREPROCESSING, errors=[]
+    )
+    try:
```

Comment:
Wäre besser gekapselt, wenn die task spezifischen Sachen in eine eigene Klasse ausgelagert wären und hier instanziiert werden würden.